src/UmbracoExamine.PDF/ExaminePdfComponent.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Linq;
+using Examine;
+using Examine.LuceneEngine.Providers;
+using Lucene.Net.Index;
+using Umbraco.Core;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Core.Services.Implement;
+using Umbraco.Examine;
+using Umbraco.Web.Scheduling;
+
+namespace UmbracoExamine.PDF
+{
+    /// <summary>
+    ///     Component to index PDF documents in the media library
+    /// </summary>
+    public class ExaminePdfComponent : IComponent
+    {
+        private static bool _disableExamineIndexing = false;
+        private static bool _isConfigured = false;
+        private static object _configuredInit = null;
+        private static object _isConfiguredLocker = new object();
+        private static readonly object RebuildLocker = new object();
+        private static BackgroundTaskRunner<IBackgroundTask> _rebuildOnStartupRunner;
+
+        private readonly IExamineManager _examineManager;
+        private readonly IndexRebuilder _indexRebuilder;
+        private readonly ILogger _logger;
+        private readonly PdfIndexCreator _pdfIndexCreator;
+        private readonly PdfIndexPopulator _pdfIndexPopulator;
+
+
+        public ExaminePdfComponent(
+            IExamineManager examineManager,
+            PdfIndexCreator pdfIndexCreator,
+            ILogger logger,
+            PdfIndexPopulator pdfIndexPopulator,
+            IndexRebuilder indexRebuilder)
+        {
+            _examineManager = examineManager;
+            _pdfIndexCreator = pdfIndexCreator;
+            _logger = logger;
+            _pdfIndexPopulator = pdfIndexPopulator;
+            _indexRebuilder = indexRebuilder;
+        }
+
+        public void Initialize()
+        {
+            foreach (var index in _pdfIndexCreator.Create())
+            {
+                // ensure the index is unlocked 
+                if (index is LuceneIndex luceneIndex)
```

Comment:
Aha, ok i can see why this seems like it needs to be explicitly done. It's because Umbraco does unlock all lucene indexes but this could occur before this component executes since Umbraco does that in it's own ExamineComponent. I now realize there is an ordering thing that needs to happen for custom index implementations. We'll need to update Umbraco itself to help with this. In the interim we'll need to manually unlock :(

Reply:
Have added TODOs in there so we can remove this when the lib is updated to reference 8.1.2